api/rest/restapi.go

```diff
@@ -379,6 +379,18 @@ func (api *API) routes() []route {
 			"/pins/{hash}",
 			api.unpinHandler,
 		},
+		{
+			"PinPath",
+			"POST",
+			"/pins/{keyType:ipfs|ipns|ipld}/{path:.*}",
```

Comment:
may be also support paths without keytype? In that case keytype would be `ipfs`

Reply:
Right now the existing `/pins/<cid>` is equivalent to `pins/ipfs/<cid>`, but other than that I prefer not, it creates some ambiguity.

Reply:
`/<cid>/path/to/file` is also a valid path, probably we can support that too. then we will need just one regex `/pins/{path:.*}`
https://github.com/ipfs/go-path/blob/2b7faf21c235d566731ee484cf2aad6a3fe3ff36/path.go#L22-L27